src/Compilers/CSharp/Test/Symbol/DeclarationTests.cs

```diff
@@ -355,6 +355,529 @@ public class b
             Assert.Equal(1, countedTree.AccessCount);
         }
 
+        [Fact]
+        public void TestMemberNamesReused1()
+        {
+            var firstTree = SyntaxFactory.ParseSyntaxTree(@"
+using System;
+
+namespace N1
+{
+    namespace N2.N3
+    {
+        class C
+        {
+            int x, y, z;
+        }
+    }
+}
+");
+            var secondTree = SyntaxFactory.ParseSyntaxTree(@"
+using System;
+
+namespace N1
+{
+    namespace N2.N3
+    {
+        class C
+        {
+            int z, y, x;
+        }
+    }
+}
+");
+            var thirdTree = SyntaxFactory.ParseSyntaxTree(@"
+using System;
+
+namespace N1
+{
+    namespace N2.N3
+    {
+        class C
+        {
+            int z, y, x, w;
+        }
+    }
+}
+");
+
+            var compilation = CreateCompilation(new SyntaxTree[] { firstTree });
+
+            var type1 = (SourceNamedTypeSymbol)compilation.GetTypeByMetadataName("N1.N2.N3.C");
+            Assert.True(type1.MergedDeclaration.Declarations[0].MemberNames.Value.SetEquals(new[] { "x", "y", "z" }));
+
+            compilation = compilation.ReplaceSyntaxTree(firstTree, secondTree);
+
+            var type2 = (SourceNamedTypeSymbol)compilation.GetTypeByMetadataName("N1.N2.N3.C");
+            Assert.True(type2.MergedDeclaration.Declarations[0].MemberNames.Value.SetEquals(new[] { "x", "y", "z" }));
+
+            // We should have the exact same set for the names.
+            Assert.True(type1.MergedDeclaration.Declarations[0].MemberNames == type2.MergedDeclaration.Declarations[0].MemberNames);
```

Comment:
>True [](http://example.com/codeflow?start=19&length=4)

```Same```? Here and in other similar places.

Reply:
> ```Same```? Here and in other similar places.

It looks like there was no response to this comment

Reply:
Done.